dot/parachain/provisioner/prioritized_disputes_selection.go

```diff
@@ -15,10 +17,236 @@ import (
 	"github.com/ChainSafe/gossamer/lib/primitives"
 )
 
+const (
+	MaxDisputeVotesForwardedToRuntime = 200_000
+	VotesSelectionBatchSize           = 1_100
+)
+
 type BlockState interface {
 	GetRuntime(blockHash common.Hash) (instance parachain.RuntimeInstance, err error)
 }
 
+// Implements the `select_disputes` function which selects dispute votes which should
+// be sent to the Runtime.
+//
+// Provisioner fetches all disputes from `dispute-coordinator` and separates them in multiple
+// partitions. Please refer to `struct PartitionedDisputes` for details.
+//
+// Besides the prioritization described above the votes in each partition are filtered too.
+// Provisioner fetches all onchain votes and filters them out from all partitions. As a result the
+// Runtime receives only fresh votes
+func SelectDisputes(
+	overseerChan chan<- any,
```

Comment:
I would hope we could use a minimal interface that all expected messages would conform to for the `overseerChan`.